packages/common/infra/src/orm/core/adapters/yjs/table.ts

```diff
@@ -104,36 +104,54 @@ export class YjsTableAdapter implements TableAdapter {
     const { where, select, callback } = query;
 
     let listeningOnAll = false;
-    const obKeys = new Set<any>();
-    const results = [];
+    const resultKeys = new Map<string, number>();
+    const results: any[] = [];
 
     if (!where) {
       listeningOnAll = true;
-    } else if ('byKey' in where) {
-      obKeys.add(where.byKey.toString());
     }
 
     for (const record of this.iterate(where)) {
-      if (!listeningOnAll) {
-        obKeys.add(this.keyof(record));
-      }
+      resultKeys.set(this.keyof(record), results.length);
       results.push(this.value(record, select));
     }
 
     callback(results);
 
     const ob = (tx: Transaction) => {
+      let hasChanged = false;
       for (const [ty] of tx.changed) {
-        const record = ty as unknown as AbstractType<any>;
-        if (
-          listeningOnAll ||
-          obKeys.has(this.keyof(record)) ||
-          (where && this.match(record, where))
-        ) {
-          callback(this.find({ where, select }));
-          return;
+        const record = ty;
+        const key = this.keyof(record);
+        const isMatch =
+          (listeningOnAll || (where && this.match(record, where))) &&
+          !this.isDeleted(record);
+        const prevMatchedIndex = resultKeys.get(key) ?? -1;
+        const prevMatch =
+          prevMatchedIndex !== -1 ? results[prevMatchedIndex] : null;
+        const isPrevMatched = prevMatchedIndex !== -1;
+
+        if (isMatch && isPrevMatched) {
+          const newValue = this.value(record, select);
+          if (prevMatch !== newValue) {
+            results[prevMatchedIndex] = newValue;
+
+            hasChanged = true;
+          }
+        } else if (isMatch && !isPrevMatched) {
+          resultKeys.set(this.keyof(record), results.length);
+          results.push(this.value(record, select));
+          hasChanged = true;
+        } else if (!isMatch && isPrevMatched) {
+          resultKeys.delete(key);
+          results.splice(prevMatchedIndex, 1);
```

Comment:
this is wrong, if a record in front of the result get deleted, the following elements will be move to a index that differs from the stored value in `resultKeys`